### PR TITLE
Add refund/owed XML

### PR DESCRIPTION
**Pull request by embarnard**

### CODE REVIEW

File: app/lib/submission_builder/ty2022/states/az/documents/financial_transaction.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module SubmissionBuilder
+  module Ty2022
+    module States
+      module Az
+        module Documents
+          class FinancialTransaction < SubmissionBuilder::Document
+            include SubmissionBuilder::FormattingMethods
+
+            def document
+              build_xml_doc("efile:FinancialTransaction") do |xml|
```
Comment (embarnard) [inline app/lib/submission_builder/ty2022/states/az/documents/financial_transaction.rb:10]: Screenshot 2023-11-20 at 4 15 19 PM
seeing this error when I try to include financial transaction node for AZ

File: app/lib/submission_builder/ty2022/states/ny/documents/rtn_header.rb [hunk 2]
```diff
@@ -30,12 +37,15 @@ def document
...
                 end
-                # xml.PYMT_AMT
+                unless @submission.data_source.withdraw_amount.nil?
+                  xml.PYMT_AMT claimed: @submission.data_source.withdraw_amount
+                end
                 # xml.DCMT_RCVD_DT
                 # xml.PSTMRK_DT
-                xml.ACH_IND
-                xml.RFND_OWE_IND
+                xml.ACH_IND claimed: 2 # no ACH debit block
```
Comment (embarnard) [inline app/lib/submission_builder/ty2022/states/ny/documents/rtn_header.rb:45]: only for taxes owed for direct debit

File: app/models/state_file_ny_intake.rb [hunk 3]
```diff
@@ -153,6 +153,17 @@ def calculated_refund_or_owed_amount
...
   end
 
+  def refund_or_owe_taxes_type
+    amt = calculated_refund_or_owed_amount
+    if amt == 0
+      :none
+    elsif amt > 0
+      :refund
+    elsif amt < 0
+      :owe
```
Comment (mpidcock) [inline app/models/state_file_ny_intake.rb:163]: Passing on these suggestions from rubocop, idk if it matters 

```suggestion
    if amt.zero?
      :none
    elsif amt.positive?
      :refund
    elsif amt.negative?
      :owe
```

File: app/lib/submission_builder/ty2022/states/ny/documents/rtn_header.rb [hunk 4]
```diff
@@ -30,12 +36,13 @@ def document
...
-                # xml.PYMT_AMT
-                # xml.DCMT_RCVD_DT
-                # xml.PSTMRK_DT
-                xml.ACH_IND
-                xml.RFND_OWE_IND
+                unless @submission.data_source.withdraw_amount.nil?
+                  xml.PYMT_AMT claimed: @submission.data_source.withdraw_amount
+                end
+                xml.ACH_IND claimed: @submission.data_source.ach_debit_transaction? ? 1 : 2
+                xml.RFND_OWE_IND claimed:  REFUND_OR_OWE_TYPES[@submission.data_source.refund_or_owe_taxes_type]
```
Comment (mpidcock) [inline app/lib/submission_builder/ty2022/states/ny/documents/rtn_header.rb:43]: There's an extra space here
```suggestion
                xml.RFND_OWE_IND claimed: REFUND_OR_OWE_TYPES[@submission.data_source.refund_or_owe_taxes_type]
```

**Review by mpidcock — APPROVED** [review]
lgtm, you can ignore my nitpicking if you want

**Pull request merged**